features/AddKudo.go

```diff
@@ -16,30 +16,31 @@ type AddKudo struct {
 	kudoCounts model.KudoCounts
 }
 
+var lastUpdate = make(map[int64]tgbotapi.Update)
+
 func NewAddKudo(kudosService model.Kudos, userService model.Users, kudoCountService model.KudoCounts) AddKudo {
+
 	return AddKudo{kudos: kudosService, users: userService, kudoCounts: kudoCountService}
 }
 
 func (k AddKudo) Execute(update tgbotapi.Update, db *gorm.DB, bot *tgbotapi.BotAPI) {
+	if lastUpdate[update.Message.Chat.ID].Message == nil {
+		return
+	}
 	var msg tgbotapi.MessageConfig
-	receiver, err := k.users.CreateUserIfNotExist(update.Message.ReplyToMessage.From, db)
+	_, err := k.users.CreateUserIfNotExist(update.Message.From, db)
 	if err != nil && err != gorm.ErrRecordNotFound {
 		msg = tgbotapi.NewMessage(update.Message.Chat.ID, fmt.Sprintf("Kudo error: %s", err))
 	}
 
-	_, err = k.users.CreateUserIfNotExist(update.Message.From, db)
+	receiver, err := k.users.CreateUserIfNotExist(lastUpdate[update.Message.Chat.ID].Message.From, db)
+	if err != nil && err != gorm.ErrRecordNotFound {
```

Comment:
ErrRecordNotFound is hier niet mogelijk toch? Want als die voorkomt wordt er een nieuwe user gemaakt.